pandas/io/xml.py

```diff
@@ -413,11 +413,21 @@ def _iterparse_nodes(self) -> list[dict[str, str | None]]:
                     row = {}
 
             if row is not None:
-                for col in self.iterparse[row_node]:
-                    if curr_elem == col:
-                        row[col] = elem.text.strip() if elem.text else None
-                    if col in elem.attrib:
-                        row[col] = elem.attrib[col]
+                if self.names:
```

Comment:
Could you share this?

Reply:
Good point. I refactored to use code once in base class of the parsers.

Reply:
This is a bit confusing now. Could you split the refactoring and the bug fix into 2 prs? Hard to review like this

Reply:
Will revert back to bug fix only and on subsequent PR refactor to remove repetitive code between parsers.